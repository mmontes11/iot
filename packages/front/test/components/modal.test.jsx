import React from "react";
import { shallow } from "enzyme";
import Modal from "components/modal";

describe("components/modal", () => {
  const onCloseClick = jest.fn();
  const wrapper = shallow(
    <Modal isActive onCloseClick={onCloseClick} messageStyle="is-info" title="Test" subTitle="Test" />,
  );
  it("renders when is active", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".modal").hasClass("is-active")).toBeTruthy();
    expect(wrapper.find(".message").hasClass("is-info")).toBeTruthy();
  });
  it("simulates a click in closing button", () => {
    wrapper.find(".modal-close").simulate("click");
    expect(onCloseClick).toHaveBeenCalledTimes(1);
  });
});
