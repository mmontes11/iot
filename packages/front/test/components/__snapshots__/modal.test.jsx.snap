// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal renders when is active 1`] = `
<div
  className="modal is-active"
>
  <div
    className="modal-background"
    onClick={[MockFunction]}
    onKeyPress={[MockFunction]}
    role="button"
    tabIndex={0}
  />
  <div
    className="modal-content"
  >
    <article
      className="message is-medium is-info"
    >
      <div
        className="message-header"
      >
        <p>
          Test
        </p>
      </div>
      <div
        className="message-body has-text-centered"
      >
        Test
      </div>
    </article>
  </div>
  <button
    aria-label="close"
    className="modal-close is-large"
    onClick={[MockFunction]}
  />
</div>
`;
